src/support.js

```diff
@@ -613,20 +613,19 @@ var jsCallStartIndex = 1;
 var functionPointers = new Array({{{ RESERVED_FUNCTION_POINTERS }}});
 #endif // WASM_BACKEND && RESERVED_FUNCTION_POINTERS
 
-// 'sig' parameter is only used on LLVM wasm backend
+// 'sig' parameter is only used on for LLVM backend with EMULATED_FUNCTION_POINTERS
 function addFunction(func, sig) {
-#if WASM_BACKEND
-  assert(typeof sig !== 'undefined',
-         'Second argument of addFunction should be a wasm function signature ' +
-         'string');
-#endif // WASM_BACKEND
-#if ASSERTIONS
+#if ASSERTIONS == 2
   if (typeof sig === 'undefined') {
     err('warning: addFunction(): You should provide a wasm function signature string as a second argument. This is not necessary for asm.js and asm2wasm, but is required for the LLVM wasm backend, so it is recommended for full portability.');
```

Comment:
this error text seems out of date. also, it should only be checked when relevant, I think, and it looks like it would be hit in asm2wasm and without emulated function pointers too?